pages/book/contracts.mdx

```diff
@@ -0,0 +1,203 @@
+const GlobalConst1: Int = 1000 + ton("42") + pow(10, 9);
+
+contract Example {
+    // contract constants are also calculated in compile-time and cannot change
+    const ContractConst1: Int = 2000 + ton("43") pow(10, 9);
+
+    // contract constants can be an easy alternative to enums
+    const StateUnpaid: Int = 0;
+    const StatePaid: Int = 1;
+    const StateDelivered: Int = 2;
+    const StateDisputed: Int = 3;
+
+    // no need to init constants
+    init() {}
+
+    get fun sum(): Int {
+        // access constants from anywhere
+        return GlobalConst1 + self.ContractConst1 + self.StatePaid;
+    }
+}
+```
+
+Read more about constants on their dedicated page: [Constants](/book/constants).
+
+### Constructor function `init()` [#init-function]
+
+On deployment of the contract, the constructor function `init(){:tact}` is run.
+
+If a contract has any [persistent state variables](#variables) without default values specified, it must initialize them in this function.
+
+As stated previously, [contract constants](#constants) don't need to be initialized in the `init{:tact}` function, as they get their values in _compile-time_ and don't consume space in persistent state.
+
+```tact
+contract Example {
+    // persistent state variables
+    var1: Int = 0; // initialized with default value 0
+    var2: Int;     // must be initialized in the init() function
+
+    // constructor function
+    init() {
+        self.var2 = 42;
+    }
+}
+```
+
+<Callout>
+
+  To obtain initial state of the target contract in [internal functions](#internal-functions), [receivers](#receiver-functions) or [getters](#getter-functions) use [`initOf{:tact}`](/book/expressions#initof) expression.
+
+</Callout>
+
+### Getter functions
+
+Getter functions are **not accessible from other contracts and exported only to off-chain world**:
+
+```tact
+contract HelloWorld {
+    counter: Int;
+
+    init() {
+        self.counter = 0;
+    }
+
+    receive("increment") {
+        self.counter += 1;
+    }
+
+    // getter function with return type Int
+    get fun counter(): Int {
+        return self.counter;
+    }
+}
+```
+
+Read more about them in their dedicated section: [Functions](/book/functions#getter-functions)
+
+### Receiver functions
+
+Receiver functions in Tact can be one of three kinds:
+
+* [`receive(){:tact}`](/book/receive), which receive internal messages (from other contracts).
+* [`bounced(){:tact}`](/book/bounced), which are called when outgoing message from this contract has bounced back.
+* [`external(){:tact}`](/book/bounced), which don't have a sender and can be sent by anyone in the world.
+
+```tact
+contract HelloWorld {
+    counter: Int;
+
+    init() {
+        self.counter = 0;
+    }
+
+    // internal message receiver, which responds to a string message "increment"
+    receive("increment") {
+        self.counter += 1;
+    }
+
+    get fun counter(): Int {
+        return self.counter;
+    }
```

Comment:
I would expect to see at least `bounced` in this example. What about `external`?